python/taichi/lang/kernel_impl.py

```diff
@@ -844,7 +846,9 @@ def __call__(self, *args, **kwargs):
         # Both the class kernels and the plain-function kernels are unified now.
         # In both cases, |self.grad| is another Kernel instance that computes the
         # gradient. For class kernels, args[0] is always the kernel owner.
-        if self.autodiff_mode == AutodiffMode.NONE and self.runtime.target_tape and not self.runtime.grad_replaced:
+
+        # No need to capture grad kernels because they are already binded with their primal kernels
+        if self.autodiff_mode != AutodiffMode.REVERSE and self.runtime.target_tape and not self.runtime.grad_replaced:
```

Comment:
```suggestion
        if self.autodiff_mode in (AutodiffMode.NONE, AutodiffMode.VALIDATION) and self.runtime.target_tape and not self.runtime.grad_replaced:
```

Reply:
Let's be more explicit here in case we add a new autodiff mode in the future (to prevent it being accidentally included ;)